src/c-writer.cc

```diff
@@ -1104,10 +1117,18 @@ void CWriter::WriteTags() {
   }
 }
 
+bool CWriter::SkipImport(const Import* import) const {
+  return options_.no_sandbox && import->module_name == kEnvModuleName &&
+         import->field_name == kStackPointerNameWithoutDollar;
```

Comment:
Why not just leave this as a normal global import?   Then it could be defined by the runtime like any other external/import the module needs?

Also, the stack pointers is shared between all the instances,  so I think you want to make sure you don't define a different stack for each module.

Reply:
In other words, I think it would be really nice if we could make this change without modifying the c-writer.cc, but instead modifying he surrounding / embedding code?  Maybe there is some fundamental reason this doesn't work?

Reply:
we are going to loose module instance structure. I should probably reorder these commits...

(also this PR is apparently broken. I might have done wrong cherry-picks, the tests are not passing atm)

Reply:
done.